Feature/UserInfoController

VladyslavCherednychenko opened the pull request:
add UserInfoController with possibility to show and update information about current authorized user 
add tokenLifeSpan variable in LoginController to facilitate interaction with token expiration time 
add more test entities to increase the information obtained in testing 

make ef migration with additionally created entities 

change HttpPut to HttpPost in UpdateEmployee request from EmployeeController 

convert to block scoped namespace in TeamsController

VladyslavCherednychenko committed:
Feature/UserInfoController

add UserInfoController with possibility to show and update information about current authorized user
add tokenLifeSpan variable in LoginController to facilitate interaction with token expiration time
add more test entities to increase the information obtained in testing
make ef migration with additionally created entities
change HttpPut to HttpPost in UpdateEmployee request from EmployeeController
convert to block scoped namespace TeamsController

VladyslavCherednychenko committed:
add missed EmployeeUserUpdateDTO

dangoth commented on Portal_Perfomance_Employes/Controllers/EmployeeController.cs:
```diff
@@ -31,7 +32,7 @@ public async Task<IActionResult> GetEmployee(int id)
-        
+
         [HttpPost]
         [Authorize(Roles = "Admin")]
```
If we annotate the whole controller as requiring authorize, then we don't need the annotations for individual methods

VladyslavCherednychenko replied:
Yep, In a new commit I moved function GetEmployee from that controller to UserInfoController and make this controller only for admins

dangoth commented on Portal_Perfomance_Employes/Controllers/UserInfoController.cs:
```diff
@@ -0,0 +1,97 @@
+        [HttpGet]
+        public IActionResult GetUserRole()
+        {
+            return Ok(User.FindFirst(ClaimTypes.Role)?.Value);
```
FindFirst throws an exception if no results are found, so I think we should try catch that if we want to use it. Otherwise we could use FirstOrDefault

VladyslavCherednychenko replied:
Corrected in a new commit

VladyslavCherednychenko committed:
fix issue with authorization in AuthResponsesOperationFilter
fix issue with crash when querying users without roles in UserInfoController
move function GetEmployee from EmployeeController to UserInfoController
rename function GetEmployee to GetEmployeeProfile
add a filter for the function GetEmployeeProfile

DmytroDorodnykhTeamInternational commented on Portal_Perfomance_Employes/Controllers/UserInfoController.cs:
```diff
@@ -0,0 +1,150 @@
+            }
+            else if (emp?.Role == null)
+            {
+
```
spacing

VladyslavCherednychenko replied:
It will be done

DmytroDorodnykhTeamInternational commented on Portal_Perfomance_Employes/Controllers/UserInfoController.cs:
```diff
@@ -0,0 +1,150 @@
+        public async Task<IActionResult> GetUserTeamHistory()
+        {
+            int.TryParse(User.FindFirst(ClaimTypes.NameIdentifier)?.Value, out int id);
+            var emp = await _context.TeamMembers.Where(e => e.EmployeeId == id).ToListAsync();
```
can't you get it from user itself?

VladyslavCherednychenko replied:
No, user has a ClaimsPrincipal type, and it is for authorization, so I have no possibility to get db context from it

DmytroDorodnykhTeamInternational commented on Portal_Perfomance_Employes/Controllers/UserInfoController.cs:
```diff
@@ -0,0 +1,150 @@
+                }
+                else
+                {
+                    var empTeam = await _context.TeamMembers.FirstOrDefaultAsync(e => e.EmployeeId == empId && e.IsActive);
```
same

VladyslavCherednychenko committed:
fix spacing

VladyslavCherednychenko committed:
resolve conflict in AppDbContext

VladyslavCherednychenko committed:
Merge branch 'development' into feature/add_UserInfoController
